pkg/cmd/readyz/readyz.go

```diff
@@ -0,0 +1,187 @@
+	keepaliveTimeout  = 10 * time.Second
+	DefaultListenPort = 7443
+)
+
+type readyzOpts struct {
+	listenPort       uint16
+	dialTimeout      time.Duration
+	targetEndpoint   string
+	clientCertFile   string
+	clientKeyFile    string
+	clientCACertFile string
+}
+
+// NewReadyzCommand creates a readyz command that runs as an http-get readiness server alongside the etcd member container
+func NewReadyzCommand() *cobra.Command {
+	opts := readyzOpts{
+		listenPort: uint16(DefaultListenPort),
+	}
+	cmd := &cobra.Command{
+		Use:   "readyz",
+		Short: "Serve the HTTP /readyz endpoint health check for an etcd member",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := opts.Validate(); err != nil {
+				klog.Fatal(err)
+			}
+			if err := opts.Run(); err != nil {
+				klog.Fatal(err)
+			}
+		},
+	}
+
+	opts.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (r *readyzOpts) AddFlags(fs *pflag.FlagSet) {
+	fs.Uint16Var(&r.listenPort, "listen-port", r.listenPort, "Listen on this port. Default 7443")
+	fs.DurationVar(&r.dialTimeout, "dial-timeout", defaultHTTPDialTimeout, "Dial timeout for the client. Default 2s")
+	fs.StringVar(&r.targetEndpoint, "target", defaultEndpoint, "Target endpoint to perform health check against. Default https://localhost:2379")
+	fs.StringVar(&r.clientCertFile, "cert-file", "", "Health probe TLS client certificate file. (required)")
+	fs.StringVar(&r.clientKeyFile, "key-file", "", "Health probe TLS client key file. (required)")
+	fs.StringVar(&r.clientCACertFile, "cacert-file", "", "Health probe TLS client CA certificate file. (required)")
+}
+
+// Validate verifies the inputs.
+func (r *readyzOpts) Validate() error {
+	if len(r.targetEndpoint) == 0 {
+		return fmt.Errorf("missing required flag: --target")
+	}
+	if len(r.clientKeyFile) == 0 {
+		return errors.New("missing required flag: --key-file")
+	}
+	if len(r.clientCertFile) == 0 {
+		return errors.New("missing required flag: --cert-file")
+	}
+	if len(r.clientCACertFile) == 0 {
+		return errors.New("missing required flag: --cacert-file")
+	}
+
+	return nil
+}
+
+// Run contains the logic of the readyz command which checks the health of the etcd member
+func (r *readyzOpts) Run() error {
+	shutdownCtx, cancel := context.WithCancel(context.Background())
+	shutdownHandler := server.SetupSignalHandler()
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/readyz", r.getReadyzHandlerFunc(shutdownCtx))
+
+	addr := fmt.Sprintf("0.0.0.0:%d", r.listenPort)
+	klog.Infof("Listening on %s", addr)
+
+	server := &http.Server{
+		Addr:        addr,
+		Handler:     mux,
+		BaseContext: func(_ net.Listener) context.Context { return shutdownCtx },
+	}
+	go func() {
+		defer cancel()
+		<-shutdownHandler
+		klog.Infof("Received SIGTERM or SIGINT signal, shutting down server.")
+		server.Shutdown(shutdownCtx)
+	}()
+
+	c := net.ListenConfig{}
+	c.Control = permitAddressReuse
+	ln, err := c.Listen(shutdownCtx, "tcp", addr)
+	if err != nil {
+		return err
+	}
+	err = server.Serve(ln)
```

Comment:
Just realized this ought to be `ServeTLS()` since the guard controller pods are using HTTPS
```YAML
      readinessProbe:
        httpGet:
          path: healthz
          port: 7443
          host: 10.0.153.60
          scheme: HTTPS
```